react/src/pages/AccountsComponent.js

```diff
@@ -19,390 +44,329 @@ function isCurrentAccount(header, info) {
     return false;
 }
 
-const AccountInfoLeft = ({ header, info, foursightUrl }) => {
+const PortalAccessKeyStatus = ({ portalAccessKeyResponse }) => {
+    if (!portalAccessKeyResponse) {
+        return <>&ndash;</>
+    }
+    else if (portalAccessKeyResponse.invalid) {
+        return <b style={{color:"red"}}>Invalid</b>
+    }
+    else if (portalAccessKeyResponse.expired) {
+        return <b style={{color:"red"}}>Expired</b>
+    }
+    else if (portalAccessKeyResponse.expires_at) {
+        if (portalAccessKeyResponse.expires_soon) {
+            return <span style={{color:"red"}}>Expires {Time.FromNow(portalAccessKeyResponse.expires_at)} {Char.RightArrow} {portalAccessKeyResponse.expires_at}</span>
+        }
+        else {
+            return <>Expires {Time.FromNow(portalAccessKeyResponse.expires_at)} {Char.RightArrow} {portalAccessKeyResponse.expires_at}</>
+        }
+    }
+    else {
+        return <>No expiration</>
+    }
+}
 
-    const SslCertificateLink = ({ url }) => {
-        return url && url.startsWith("https://") && <small>
-          &nbsp;| <a style={{color:"inherit"}} href={Client.Path("certificates") + "/?hostname=" + url} rel="noreferrer" target="_blank"><b>SSL</b></a>&nbsp;
-          <a style={{color:"inherit"}} href={Client.Path("certificates") + "/?hostname=" + url} rel="noreferrer" target="_blank"><span className="fa fa-external-link" style={{position:"relative",bottom:"-1px"}}></span></a>
-        </small>
+const SslCertificateLink = ({ url }) => {
+    const sslCertificate = useFetch(url ? `/certificates?hostname=${url}` : null);
+    function sslCertificateExpiresSoon() {
+        return Type.IsArray(sslCertificate.data) ? sslCertificate.data[0]?.expires_soon : null;
+    }
+    function sslCertificateExpiresMessage() {
+        if (!Type.IsArray(sslCertificate.data)) return "";
+        const invalid = sslCertificate.data[0]?.invalid;
+        const expired = sslCertificate.data[0]?.expired;
+        const expiresSoon = sslCertificate.data[0]?.expired;
+        const expiresAt = sslCertificate.data[0]?.expires_at;
+        if (invalid) {
+            return `SSL certificate is invalid!`;
+        }
+        else if (expired) {
+            return `SSL certificate expired ${Time.Ago(expiresAt)} (${expiresAt})`;
+        }
+        else if (expiresSoon) {
+            return `SSL certificate expires ${Time.FromNow(expiresAt)} (${expiresAt}) -> Soon!`;
+        }
+        else {
+            return `SSL certificate expires ${Time.FromNow(expiresAt)} (${expiresAt})`;
+        }
     }
+    const uuid = Uuid();
+    return url && url.startsWith("https://") && <small>
+        <small style={{marginLeft:"3pt",marginRight:"3pt"}}>|</small>
```

Comment:
Not to fix in this moment, but I just want to say I'd so much rather you have a style sheet somewhere that defines some standard presentation modes and then use use `<small class='...'>` or some more abstract way of packaging it entirely, rather than detailing the style details inline everywhere.